packages/@storefront/core/src/core/url-beautifier/details/parser.ts

```diff
@@ -18,13 +18,10 @@ export class DetailsUrlParser extends UrlParser<UrlBeautifier.DetailsUrlState> {
     const title = path.shift();
     const id = path.pop();
     let variants = [];
-    const queryParams: {[key: string]: string} = {};
-
-    if (Object.keys(uri.query).length) {
-      const filteredQueryParams = DetailsUrlParser.extractFilteredQueryParams(uri.query);
-      queryParams.area = filteredQueryParams.area;
-      queryParams.collection = filteredQueryParams.collection;
-    }
+    const queryParams: { [key: string]: string } = {
+      ...( uri.query.area ? { area: uri.query.area } : {} ),
+      ...( uri.query.collection ? { collection: uri.query.collection } : {} )
```

Comment:
I'm not sure how I feel about this. I would have just done something like the following:

```js
const queryParams: { [key: string]: string } = {};
const { area, collection } = uri.query;

if (area) queryParams.area = area;
if (collection) queryParams.collection = collection;
```

But if you choose to keep the spreading, remove the padding between the `()`.